lib/manageiq/appliance_console/external_auth_options.rb

```diff
@@ -79,11 +80,49 @@ def update_configuration(update_hash = nil)
       if update_hash.present?
         say("\nUpdating external authentication options on appliance ...")
         params = update_hash.collect { |key, value| "#{key}=#{value}" }
+        params = never_enable_saml_and_oidc(params)
         result = ManageIQ::ApplianceConsole::Utilities.rake_run("evm:settings:set", params)
         raise parse_errors(result).join(', ') if result.failure?
       end
     end
 
+    def never_enable_saml_and_oidc(params)
+      if params.include?("/authentication/oidc_enabled=true") && params.include?("/authentication/saml_enabled=true")
```

Comment:
I'm not sure what the user experience is here, but why can this even happen? Shouldn't we ask for the user to specify only one external auth type? Or check for this as they are selecting the options rather than assuming one way or the other after the fact?

Either way I think it would be better to fail here rather than just picking one of the options they said they want to configure.

Reply:
The logic in `external_auth_options.rb` expects all the options to be a toggle of a `true` or `false` value. The new authentication setting `provider_type` is used by the UI  to toggle `oidc_enabled` and `saml_enabled`.

After speaking with @abellotti about this, the agreed approach is to get the needed changes in that will make it work even if not elegant and avoid a larger refactoring, since the fate of OIDC on non-podified appliances is not finalized.

Because the appliance_console loops allowing the user to reselect options the params array could have many combinations. Without reworking `external_auth_options.rb` the solution I presented ensures correct results.

That said, I could error out if both `oidc_enabled` and `saml_enabled` are true.

Reply:
Oh and thanks for taking a look! @carbonin

Reply:
Okay, then I think a `validate!` which will error if there is an incompatible combination selected is a better experience than possibly configuring something the user may not have intended.

Reply:
@carbonin 

The only incompatible combination it so enable both oidc and saml. I will add a validate! for that and error out.

Thanks!